Update grants to include revocation examples (#6678)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/grants.md

dbt aims to use the most efficient approach when updating grants, which varies based on the adapter you're using, and whether dbt is replacing or updating an object that already exists. You can always check the debug logs for the full set of grant and revoke statements that dbt runs.

You should define grants as resource configs whenever possible, but you might occasionally need to write grants statements manually and run them using [hooks](/docs/build/hooks-operations). For example, hooks may be appropriate if you want to:

* Apply grants on other database objects besides views and tables.
* Create more granular row- and column-level access, use masking policies, or apply future grants.

> * Take advantage of more-advanced permission capabilities offered by your data platform, for which dbt does not offer out-of-the-box support using resource configuration.

Reviewer: ```suggestion
* Take advantage of more advanced permission capabilities offered by your data platform, for which dbt does not offer out-of-the-box support using resource configuration.
```

> * Apply grants in a more complex or custom manner, beyond what the built-in grants capability can provide.

Author: This is not a change in content, just reordering to lead with the more specific examples and have the "etc" ones come second.

- This use of `+`, controlling clobber vs. add merge behavior, is distinct from the use of `+` in `dbt_project.yml` (shown in the example above) for defining configs with dictionary values. For more information, see [the plus prefix](https://docs.getdbt.com/reference/resource-configs/plus-prefix).
- `grants` is the first config to support a `+` prefix for controlling config merge behavior. Currently, it's the only one. If it proves useful, we may extend this capability to new and existing configs in the future.

### Conditional grants

Like any other config, you can use Jinja to vary the grants in different contexts. For example, you might grant different permissions in prod than dev:

<File name='dbt_project.yml'>

```yml
models:
  +grants:
    select: "{{ ['user_a', 'user_b'] if target.name == 'prod' else ['user_c'] }}"
```

</File>

## Revoking grants

> dbt will only modify grants on a node (including revocation) when a `grants` configuration is attached to that node. For example, imagine you had originally specified the following grants in `dbt_project.yml`:

Reviewer: ```suggestion
dbt only modifies grants on a node (including revocation) when a `grants` configuration is attached to that node. For example, imagine you had originally specified the following grants in `dbt_project.yml`:
```

<File name='dbt_project.yml'>

```yml
models:
  +grants:
    select: ['user_a', 'user_b']
```

</File>

> If you delete the `+grants` section altogether, dbt will assume you no longer want it to manage grants, and will not change anything. To have dbt revoke all existing grants from a node, provide an empty list of grantees instead.

Reviewer: ```suggestion
If you delete the entire `+grants` section, dbt assumes you no longer want it to manage grants and doesn't change anything. To have dbt revoke all existing grants from a node, provide an empty list of grantees.
```
